### PR TITLE
[Enhancement] Support update BE config using update statement

### LINKED ISSUE #18382: [Enhancement] Support set BE config by sql statement
## Enhancement

Currently, BE config can only be modified by be.conf file or BE web API. It would be great to support set BE config by sql statement, so it's easier to debug/manager/write test case.

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #18382

## Problem Summary(Required) ：

Currently, BE config can only be modified by be.conf file or BE web API.  This PR supports set BE config by sql statement, so it's easier to debug/manager/write test case.

```
mysql> select * from be_configs where name = 'txn_info_history_size';
+-------+-----------------------+-------+
| BE_ID | NAME                  | VALUE |
+-------+-----------------------+-------+
| 10004 | txn_info_history_size | 20000 |
+-------+-----------------------+-------+
1 row in set (0.05 sec)

mysql> update be_configs set value = cast(cast(value as bigint)*2 as string) where name = 'txn_info_history_size';
Query OK, 0 rows affected (0.07 sec)

mysql> select * from be_configs where name = 'txn_info_history_size';
+-------+-----------------------+-------+
| BE_ID | NAME                  | VALUE |
+-------+-----------------------+-------+
| 10004 | txn_info_history_size | 40000 |
+-------+-----------------------+-------+
1 row in set (0.01 sec)

```
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 39 / 65 (60.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|1|00.00%|[607]|
|:large_blue_circle:|com/starrocks/planner/SchemaTableSink.java|0|16|00.00%|[28, 29, 30, 34, 35, 36, 37, 42, 43, 44, 45, 46, 47, 52, 57, 62]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|0|1|00.00%|[71]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|5|40.00%|[1425, 1452, 1479]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|1|2|50.00%|[62]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|15|19|78.95%|[111, 112, 113, 114]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|7|7|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 39 / 65 (60.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|1|00.00%|[607]|
|:large_blue_circle:|com/starrocks/planner/SchemaTableSink.java|0|16|00.00%|[28, 29, 30, 34, 35, 36, 37, 42, 43, 44, 45, 46, 47, 52, 57, 62]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|0|1|00.00%|[71]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|5|40.00%|[1425, 1452, 1479]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|1|2|50.00%|[62]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|15|19|78.95%|[111, 112, 113, 114]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|7|7|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 39 / 65 (60.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|1|00.00%|[607]|
|:large_blue_circle:|com/starrocks/planner/SchemaTableSink.java|0|16|00.00%|[28, 29, 30, 34, 35, 36, 37, 42, 43, 44, 45, 46, 47, 52, 57, 62]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|0|1|00.00%|[71]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|5|40.00%|[1425, 1452, 1479]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|1|2|50.00%|[62]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|15|19|78.95%|[111, 112, 113, 114]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|7|7|100.00%|[]|

**Comment by decster** [comment]
run starrocks_be_unittest

File: gensrc/thrift/DataSinks.thrift [hunk 1]
```diff
@@ -197,6 +198,11 @@ struct TOlapTableSink {
     24: optional i32 auto_increment_slot_id;
 }
 
+struct TSchemaTableSink {
+    1: required string table
```
Comment (imay) [inline gensrc/thrift/DataSinks.thrift:202]: all fields should be optional.

File: gensrc/proto/internal_service.proto [hunk 2]
```diff
@@ -495,6 +495,17 @@ message PMVMaintenanceTaskResult {
     required StatusPB status = 1;
 };
 
+message PExecuteCommandRequest {
```
Comment (imay) [inline gensrc/proto/internal_service.proto:498]:
```suggestion
message ExecuteCommandRequestPB {
```

File: gensrc/proto/internal_service.proto [hunk 3]
```diff
@@ -495,6 +495,17 @@ message PMVMaintenanceTaskResult {
...
+    optional string command = 1;
+    optional string params_json = 2;
+};
+
+message PExecuteCommandResult {
```
Comment (imay) [inline gensrc/proto/internal_service.proto:503]:
```suggestion
message ExecuteCommandResultPB {
```

File: gensrc/proto/internal_service.proto (hunk 2 above)
Comment (decster) [reply gensrc/proto/internal_service.proto:498]: all other messages in this file  use Pxxx convention, are you sure that we change to PB suffix?

File: be/src/http/action/update_config_action.cpp [hunk 4]
```diff
@@ -101,6 +101,19 @@ void UpdateConfigAction::handle(HttpRequest* req) {
         });
     });
 
+    Status s = config::set_config(name, value);
```
Comment (luohaha) [inline be/src/http/action/update_config_action.cpp:104]: Make the config change persistent? It seems not persistent now.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 39 / 65 (60.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|1|00.00%|[607]|
|:large_blue_circle:|com/starrocks/planner/SchemaTableSink.java|0|16|00.00%|[28, 29, 30, 34, 35, 36, 37, 42, 43, 44, 45, 46, 47, 52, 57, 62]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|0|1|00.00%|[71]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|5|40.00%|[1425, 1452, 1479]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|1|2|50.00%|[62]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|15|19|78.95%|[111, 112, 113, 114]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|7|7|100.00%|[]|

File: be/src/http/action/update_config_action.cpp (hunk 4 above)
Comment (decster) [reply be/src/http/action/update_config_action.cpp:104]: currently, there is no way to make it persistent, the config is stored in some config file, it's not easy to re-edit config file.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 39 / 65 (60.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|1|00.00%|[607]|
|:large_blue_circle:|com/starrocks/planner/SchemaTableSink.java|0|16|00.00%|[28, 29, 30, 34, 35, 36, 37, 42, 43, 44, 45, 46, 47, 52, 57, 62]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|0|1|00.00%|[71]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|5|40.00%|[1425, 1452, 1479]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|1|2|50.00%|[62]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|15|19|78.95%|[111, 112, 113, 114]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|7|7|100.00%|[]|

**Comment by decster** [comment]
run starrocks_be_unittest

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 52 / 65 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|1|00.00%|[607]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|5|20.00%|[1426, 1452, 1453, 1477]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|1|2|50.00%|[62]|
|:large_blue_circle:|com/starrocks/planner/SchemaTableSink.java|10|16|62.50%|[34, 35, 36, 37, 52, 57]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|18|19|94.74%|[114]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|1|1|100.00%|[]|

**Comment by decster** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 50 / 61 (81.97%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|1|00.00%|[607]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|1|2|50.00%|[62]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|5|60.00%|[1428, 1482]|
|:large_blue_circle:|com/starrocks/planner/SchemaTableSink.java|10|16|62.50%|[34, 35, 36, 37, 52, 57]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|18|19|94.74%|[114]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|1|1|100.00%|[]|

**Comment by decster** [comment]
run starrocks_clang-format

**Comment by decster** [comment]
run starrocks_admit_test

**Comment by decster** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 50 / 61 (81.97%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|1|00.00%|[607]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|1|2|50.00%|[62]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|5|60.00%|[1428, 1482]|
|:large_blue_circle:|com/starrocks/planner/SchemaTableSink.java|10|16|62.50%|[34, 35, 36, 37, 52, 57]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|18|19|94.74%|[114]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|1|1|100.00%|[]|

File: gensrc/proto/internal_service.proto (hunk 2 above)
Comment (imay) [reply gensrc/proto/internal_service.proto]: yes, we can change them later

**Review by imay — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 51 / 62 (82.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|1|00.00%|[607]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|1|2|50.00%|[62]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|5|60.00%|[1428, 1482]|
|:large_blue_circle:|com/starrocks/planner/SchemaTableSink.java|11|17|64.71%|[34, 35, 36, 37, 53, 58]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|18|19|94.74%|[114]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|1|1|100.00%|[]|

**Comment by decster** [comment]
run starrocks_clang-format

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**